### PR TITLE
Updated Notification and Event Management Unit Tests

### LINKED ISSUE #96: Event Management Module
Implement unit tests for all event management operations for administrators and users.

**Acceptance Criteria:**

- [x] Test adding new events (valid and invalid inputs)
- [x] Test editing existing events
- [x] Test canceling events
- [x] Test event filtering and search functionality
- [x] Test date, location, and category filtering
- [x] Achieve minimum 85% code coverage for event management
- [ ] All tests documented with expected vs. actual results
- [ ] Tests integrated into GitHub Actions CI pipeline

### LINKED ISSUE #98: Notification Module
Develop unit tests for the notification/confirmation system (email and SMS delivery).

**Acceptance Criteria:**

- [x] Test email confirmation template generation
- [x] Test SMS message generation
- [x] Test notification triggers (registration, reservation, cancellation, confirmation)
- [x] Mock external email/SMS providers
- [x] Test error handling for failed notifications
- [x] Achieve minimum 85% code coverage
- [ ] All test cases and results documented

### PR DESCRIPTION
**Pull request by mia-moreau**
### Changes made to NotificationServiceTest (#98)
 - added a date check to email and SMS confirmation messages
 - added a test for SMS failure when it's the only option
 - added a test for booking cancelled fallback to SMS (checks exact message)
 - added a test for event cancelled fallback to SMS (checks exact message)
 - added missing edge case where email fails, and user has no phone

### Changes made to EventService + EventServiceTest (#96)
- moved the search by keyword logic from the EventListActivity to EventService
- added unit tests for search by keyword

at least 85% code coverage has been ensured for all related tests.

### CODE REVIEW

File: app/src/main/java/com/example/ticketnoob/service/EventService.java [hunk 1]
```diff
@@ -125,4 +125,31 @@ List<Event> applyFilters(List<Event> events, String date, String location, Strin
     private static String safeTrim(String s){
         return s == null ? "" : s.trim();
     }
+
+    public List<Event> applyKeywordSearch(List<Event> events, String keyword) {
```
Comment (catalexandre) [inline app/src/main/java/com/example/ticketnoob/service/EventService.java:129]: Logic is good

File: app/src/test/java/com/example/ticketnoob/service/EventServiceTest.java
Comment (catalexandre) [inline app/src/test/java/com/example/ticketnoob/service/EventServiceTest.java:1]: Maybe just add a test where keyword doesn't match anything and ensure returned size is zero

**Review by catalexandre — APPROVED** [review]
One request for an extra test. Good

File: app/src/test/java/com/example/ticketnoob/service/EventServiceTest.java
Comment (mia-moreau) [reply app/src/test/java/com/example/ticketnoob/service/EventServiceTest.java:1]: Added a test for this! I also removed a test that I noticed was redundant.
